File: module_beauty/src/main/java/com/gas/beauty/ui/classify/mvp/ClassifyContract.java
package com.gas.beauty.ui.classify.mvp;

import android.content.Context;

import com.base.lib.mvp.IView;
import com.gas.beauty.bean.GankItemBean;

import java.util.List;

import io.reactivex.Observable;

public interface ClassifyContract {


    interface View extends IView {
        void loadEnd();
        Context getWrapContext();

    }

    interface Model {
        Observable<List<GankItemBean>> getGankItemData(String suburl);
    }


}

File: module_beauty/src/main/java/com/gas/beauty/ui/classify/mvp/ClassifyContract.kt
package com.gas.beauty.ui.classify.mvp

import android.content.Context
import com.base.lib.mvp.IModel
import com.base.lib.mvp.IView
import com.gas.beauty.bean.GankItemBean
import io.reactivex.Observable

interface ClassifyContract {
    interface View : IView {
        fun loadEnd()
        val wrapContext: Context?
    }

    interface Model : IModel {
        fun getGankItemData(suburl: String): Observable<List<GankItemBean>>
    }
}